pax-web-extender-war/src/main/java/org/ops4j/pax/web/extender/war/internal/UnregisterWebAppVisitorWC.java

```diff
@@ -120,18 +120,15 @@ public void visit(final WebApp webApp) {
 	 */
 	public void visit(final WebAppServlet webAppServlet) {
 		NullArgumentException.validateNotNull(webAppServlet, "Web app servlet");
-		final Class<? extends Servlet> servletClass = webAppServlet
-				.getServletClass();
-		if (servletClass != null) {
-			//CHECKSTYLE:OFF
-			try {
-				webContainer.unregisterServlets(servletClass);
-				webAppServlet.setServletClass(null);
-			} catch (Exception ignore) { 
-				LOG.warn("Unregistration exception. Skipping.", ignore);
-			}
-			//CHECKSTYLE:ON
-		}

```

Comment:
I'm not sure about this. Actually servlets are mostly registered as Classes. This is especially important because of CDI use cases.